### ISSUE TITLE
chicViewpointBackgroundModel and `np.int` aliases deprecated in NumPy 1.20

### ISSUE THREAD

**Issue by amv33576** [deeptools/HiCExplorer#877]
Dear all,

I am finding this problem when calculating a background model. 
Command in:

chicQualityControl -m matrix.nr.h5 --referencePoints RFpoints.bed --sparsity 0.025 --threads 20 --outFileName new_RPs.bed
chicViewpointBackgroundModel -m matrix.nr.h5 --fixateRange 500000 --referencePoints new_RPs.bed --outFileName bgm.txt

ERROR:hicexplorer.chicViewpointBackgroundModel:An error occurred caused by one or many faulty reference points.
ERROR:hicexplorer.chicViewpointBackgroundModel:Please run chicQualityControl to remove these from your reference point file:  new_RPs.bed

ERROR:hicexplorer.chicViewpointBackgroundModel:module 'numpy' has no attribute 'int'.
`np.int` was a deprecated alias for the builtin `int`. To avoid this error in existing code, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

# Name               Version         Build                              Channel
hicexplorer             3.7.2            pyhdfd78af_1                  bioconda
numpy                    1.24.4          py38h59b608b_0          conda-forge
python                    3.8.18          hd12c33a_0_cpython    conda-forge

Any advice? 

Best,

A.

**Welcome to the HiCExplorer GitHub repository! Before opening the issue please check
 that the following requirements are met :**

- [ ] Search whether this issue (or a similar issue) has been solved before using the search tab above. Link the previous issue if appropriate below.

- [ ] Paste your HiCExplorer version (`hicInfo --version`) and your python version (`python --version`) below. 
- [ ] Have you checked our documentation on [hicexplorer.readthedocs.io](https://hicexplorer.readthedocs.io)?
- [ ] Do you use [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html) to install HiCExplorer? 
- [ ] Do you use the latest HiCExplorer release? If not, please install it via a conda environment: 
       `conda create --name hicexplorer hicexplorer=3.6 python=3.8 -c bioconda -c conda-forge`
       and activate the environment: `conda activate hicexplorer`.  Retry your command. You can exit a conda environment via `conda deactivate`. To learn more about conda and environments, please consider the following [documentation](https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html#).

Retry your command, is it solved now? If not please continue with the following:
- [ ] Paste the full HiCExplorer command that produces the issue below
  (ignore if you simply spotted the issue in the code/documentation).

- [ ] Paste the output printed on screen from the command that produces the issue
  below (ignore if you simply spotted the issue in the code/documentation).

**Comment by joachimwolff** [comment]
I published today a maintenance update version 3.7.3 which should fix this issue.

**Comment by robin-sto** [comment]
If someone has to use 3.5.2, how can the numpy problem be addressed?

**Comment by joachimwolff** [comment]
Go to the 3.5.2 release, and download the requirements.txt file: https://github.com/deeptools/HiCExplorer/blob/3.5.2/requirements.txt

Change now all `>=` to `==`, and install HiCExplorer using the changed requirements.txt file: `mamba create -n HiCExplorer3.5.2 --file requirements.txt hicexplorer=3.5.2`

**Comment by robin-sto** [comment]
Yes, I already have a conda env for hicexplorer3.5.2.  It shows:  numpy 1.24.4  
I get the np.int error when trying to use it.  Do you know which file in the env needs to be modified or if there is another way around it?

**Comment by joachimwolff** [comment]
Activate your environment and set the numpy version manually to an older one: `conda install numpy=1.19`

**Comment by robin-sto** [comment]
Unfortunately, there were too many conflicts to do this installation - the environment couldn't be resolved.

**Comment by joachimwolff** [comment]
You can try either mamba instead of conda, or delete your non-working environment and set it up new with the requirements.txt file as explained above.
